src/libxdata/world/models/airport/Runway.h

```diff
@@ -30,17 +30,33 @@ class Fix;
 
 class Runway: public NavNode {
 public:
+
+    enum class SurfaceType {
+        ASPHALT = 1,
+        CONCRETE = 2,
+        TURF_GRASS = 3,
+        DIRT_BROWN = 4,
+        GRAVEL_GREY = 5,
+        DRY_LAKEBED = 12,
+        WATER_RUNWAY = 13,
+        SNOW_OR_ICE = 14,
+        TRANSPARENT_SURFACE = 15
+    };
+
     Runway(const std::string &name);
     void rename(const std::string &newName);
     void setWidth(float w);
     void setLength(float l);
     void setLocation(const Location &loc);
+    void setSurfaceType(int surfaceTypeCode);

```

Comment:
The conversion (well, typecast) between the code and the enum should happen in the `AirportLoader`. The idea is that the data classes contain the raw data and numbers while the model classes contain only higher types such a enums. The loader translates the raw data to higher data, see the conversion of ATC frequency codes in `AirportLoader::onAirportLoaded` as an example.
The reason is that the model classes should be independent of the X-Plane format so that AviTab could be ported to other simulators one day. This would introduce a new loader but the model classes and thus all the apps that use them can stay the same.